File: fastfed-java-sdk/src/main/java/com/sailpoint/fastfed/sdk/providers/WhitelistProvider.java
package com.sailpoint.fastfed.sdk.providers;

import com.sailpoint.fastfed.sdk.exception.WhitelistedException;
import com.sailpoint.fastfed.sdk.model.WhitelistRecord;

public interface WhitelistProvider {
    WhitelistRecord get(String entityId) throws WhitelistedException;
    void add(WhitelistRecord whitelistRecord) throws WhitelistedException;
    void update(WhitelistRecord whitelistRecord) throws WhitelistedException;
    void remove(String entityId) throws WhitelistedException;
}

File: fastfed-node-sdk/src/interfaces/whitelist.provider.ts
import {WhitelistRecord} from "../model/whitelistRecord.model";

/**
 * Interface for a whitelisting provider to implement for some data store
 */
export interface IWhitelistProvider {
    get(entityId: string): WhitelistRecord;
    add(whitelistRecord: WhitelistRecord);
    update(whitelistRecord: WhitelistRecord);
    remove(entityId: string);
}
